File: src/main/webapp/tutorials/code/python/loops-while.py
# While Loops
# Executes a set of statements as long as a condition is true.

# 1. Basic while loop
count = 1
print("Counting up:")
while count <= 5:
    print(count)
    count += 1 # IMPORTANT: Increment to avoid infinite loop

# 2. While loop with user input simulation
# In a real app, you might use input(), but for this demo we'll simulate it
password = ""
attempts = 0
correct_password = "secret"

print("\nLogin Attempt:")
while password != correct_password and attempts < 3:
    attempts += 1
    # Simulating different inputs
    if attempts == 1:
        password = "123"
    elif attempts == 2:
        password = "secret"
    
    print(f"Attempt {attempts}: Trying '{password}'...")

if password == correct_password:
    print("Access Granted!")
else:
    print("Locked out.")

File: src/main/webapp/tutorials/code/php/loops-while.php
<?php
// PHP While Loop Demo
// 8gwifi.org/tutorials

echo "=== Basic While Loop ===\n";
$count = 1;
while ($count <= 5) {
    echo "Count: $count\n";
    $count++;
}

echo "\n=== While Loop with Array ===\n";
$fruits = ["Apple", "Banana", "Cherry", "Date"];
$index = 0;
while ($index < count($fruits)) {
    echo ($index + 1) . ". " . $fruits[$index] . "\n";
    $index++;
}

echo "\n=== While Loop: Sum Until Threshold ===\n";
$sum = 0;
$number = 1;
while ($sum < 50) {
    $sum += $number;
    echo "Added $number, Sum = $sum\n";
    $number++;
}
echo "Final sum: $sum\n";

echo "\n=== While Loop: User Input Simulation ===\n";
$attempts = 0;
$maxAttempts = 3;
$correctPassword = "secret";
$passwords = ["wrong", "incorrect", "secret"];  // Simulated inputs

while ($attempts < $maxAttempts) {
    $input = $passwords[$attempts];
    echo "Attempt " . ($attempts + 1) . ": Trying '$input'... ";

    if ($input === $correctPassword) {
        echo "Access granted!\n";
        break;
    }
    echo "Access denied.\n";
    $attempts++;
}

if ($attempts >= $maxAttempts && $passwords[$attempts-1] !== $correctPassword) {
    echo "Account locked after $maxAttempts attempts.\n";
}
?>
